tools/coqdoc/cmdArgs.ml

```diff
@@ -0,0 +1,239 @@
+                                                   inputenc = "utf8x";
+                                                   latin1 = false;
+                                                   utf8 = true } }),
+  " Set UTF-8 mode";
+  "--charset", arg_string (fun p s -> { p with encoding = { !prefs.encoding with charset = s } }),
+   "<string> Set HTML charset";
+  "--inputenc", arg_string (fun p s -> { p with encoding = { !prefs.encoding with inputenc = s } }),
+  "<string> Set LaTeX input encoding";
+  "--interpolate", arg_set (fun p -> {p with interpolate = true }),
+  " Try to typeset identifiers in comments using definitions in the same module";
+  "--raw-comments", arg_set (fun p -> {p with raw_comments = true }),
+  " Raw comments";
+  "--parse-comments", arg_set (fun p -> {p with parse_comments = true }),
+  " Parse regular comments";
+  "--plain-comments", arg_set (fun p -> {p with plain_comments = true }),
+  " Consider comments as non-literate text";
+  "--toc-depth", arg_int (fun p d -> { p with toc_depth = Some d }),
+  "<int> Don't include TOC entries for sections below level <int>";
+  "--no-lib-name", arg_set (fun p -> { p with lib_name = "" }),
+  " Don't display \"Library\" before library names in the toc";
+  "--lib-name", arg_string (fun p s -> { p with lib_name = s }),
+  "<string> Call top level toc entries <string> instead of \"Library\"";
+  "--lib-subtitles", arg_set (fun p -> { p with lib_subtitles = true }),
+  " First line comments of the form (** * ModuleName : text *) will be interpreted as subtitles";
+  "--inline-notmono", arg_set (fun p -> { p with inline_notmono = true }),
+  " Use a proportional width font for inline code (possibly with a different color)";
+  "--version", Arg.Unit (fun () -> banner()), " Display coqdoc version";
+]
+
+let deprecation_warn cmd =
+  Printf.eprintf "Warning: %s is deprecated since Coq 8.15. Please use -%s\n" cmd cmd;
+  flush stderr
+
+let add_input_files f = prefs := { !prefs with files = what_file f :: !prefs.files }
+let usage_msg = "coqdoc [options] <input file>...\nAvailable options are:"
+
+let parse_args () =
+(* Deprecated options *)
+let deprecated_opts =
+  ["-html"; "-latex"; "-texmacs"; "-raw"; "-dvi"; "-ps"; "-pdf"; "-stdout"; "-output"; "-directory"; "-gallina"; "-short"; "-light"; "-title"; "-body-only"; "-with-header"; "-with-footer"; "-no-index"; "-multi-index"; "-index"; "-toc"; "-table-of-contents"; "-vernac-file"; "-tex-file"; "-preamble"; "-files-from"; "-glob-from"; "-no-glob"; "-quiet"; "-verbose"; "-no-externals"; "-external"; "-coqlib_url"; "-coqlib"; "-latin1"; "-utf8"; "-charset"; "-inputenc"; "-interpolate"; "-raw-comments"; "-parse-comments"; "-plain-comments"; "-toc-depth"; "-no-lib-name"; "-lib-name"; "-lib-subtitles"; "-inline-notmono"; "-version"] in
+  let new_argv = Array.map (fun s -> match List.find_opt (fun m -> m = s) deprecated_opts with
+    | Some _ -> deprecation_warn s; Printf.sprintf "-%s" s
+    | None -> s) Sys.argv in
```

Comment:
I suggested that in the future we would perhaps deprecate single `-` arguments, however no such decision has been made as of yet. Please don't do this in this patch.